Management/DomainModels/Country.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Common;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Management.DomainModels
+{
+    public class Country: TaggedString<Country>
+    {
+        private enum CountryCode
```

Comment:
I feel it might not be a good idea to have a stand-alone enum class for `CountryCode`/`StateCode`.
I think all Country, State, and Location related stuff should be wrapped in the corresponding classes so it's easy to make sure that as long as an object is created, the Country/State/Location is valid. 
If we expose the enums, we have to have a separate util function for validation.

Reply:
Why so?

All mapping should happened in Mapper class which is a standalone part only response for the entity transformation. You have to reference other class anyway.

Reply:
I see your point now. Will make the enums public and move converting from enum to `Country` & `State` class to mapper.

Reply:
> I see your point now. Will make the enums public and move converting from enum to `Country` & `State` class to mapper.

Will also make the change in a separate pr.